airflow/ui/src/components/TriggerDag/TriggerDAGForm.tsx

```diff
@@ -0,0 +1,187 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Input, Button, Box, Text, Spacer, HStack } from "@chakra-ui/react";
+import { json } from "@codemirror/lang-json";
+import { githubLight, githubDark } from "@uiw/codemirror-themes-all";
+import CodeMirror from "@uiw/react-codemirror";
+import { useEffect, useState } from "react";
+import { useForm, Controller } from "react-hook-form";
+
+import { useColorMode } from "src/context/colorMode";
+
+import { Accordion } from "../ui";
+import type { DagParams } from "./TriggerDag";
+
+type TriggerDAGFormProps = {
+  dagParams: DagParams;
+  onClose: () => void;
+  onTrigger: (updatedDagParams: DagParams) => void;
+  setDagParams: React.Dispatch<React.SetStateAction<DagParams>>;
+};
+
+const TriggerDAGForm: React.FC<TriggerDAGFormProps> = ({
+  dagParams,
+  onTrigger,
+  setDagParams,
+}) => {
+  const [jsonError, setJsonError] = useState<string | undefined>();
+  const {
+    control,
+    formState: { isDirty },
+    handleSubmit,
+    reset,
+  } = useForm<DagParams>({
+    defaultValues: dagParams,
+  });
+
+  useEffect(() => {
+    reset(dagParams);
+  }, [dagParams, reset]);
+
+  const onSubmit = (data: DagParams) => {
+    onTrigger(data);
+    setDagParams(data);
+    setJsonError(undefined);
+  };
+
+  const validateAndPrettifyJson = (value: string) => {
+    try {
+      const parsedJson = JSON.parse(value) as JSON;
+
+      setJsonError(undefined);
+
+      return JSON.stringify(parsedJson, undefined, 2);
+    } catch (error) {
+      const errorMessage =
+        error instanceof Error ? error.message : "Unknown error occurred.";
+
+      setJsonError(`Invalid JSON format: ${errorMessage}`);
+
+      return value;
+    }
+  };
+
+  const { colorMode } = useColorMode();
+
+  return (
+    <>
+      <Accordion.Root collapsible size="lg" variant="enclosed">
+        <Accordion.Item key="advancedOptions" value="advancedOptions">
+          <Accordion.ItemTrigger>Advanced Options</Accordion.ItemTrigger>
+          <Accordion.ItemContent>
+            <Box p={5}>
+              <Text fontSize="md" mb={2}>
+                Logical date
+              </Text>
+              <Controller
+                control={control}
+                name="logicalDate"
+                render={({ field }) => (
+                  <Input
+                    {...field}
+                    placeholder="yyyy-mm-ddThh:mm"
+                    size="sm"
+                    type="datetime-local"
+                  />
+                )}
+              />
```

Comment:
After chatting with @uranusjr, we need to change this in the UI and in the API to optionally customize the `data_interval` instead of the logical_date directly.

Reply:
So we need to ask user both data interval start and end date?

Reply:
Yes, but that would be an API change too

Reply:
Anyways we don't have any API integration with this PR.
Will add it, can take the integration in next PR.

Reply:
Updated UI: 
<img width="914" alt="image" src="https://github.com/user-attachments/assets/c1ec074d-fc93-4487-9776-485acd2017c5">